src/gt4py/next/ffront/decorator.py

```diff
@@ -111,33 +100,14 @@ def definition(self):
 
     @functools.cached_property
     def past_stage(self):
-        if self.backend is not None and self.backend.transformer is not None:
-            return self.backend.transformer.func_to_past(self.definition_stage)
-        return next_backend.DEFAULT_TRANSFORMS.func_to_past(self.definition_stage)
+        if self.backend is not None and self.backend.transforms_prog is not None:
+            return self.backend.transforms_prog.func_to_past(self.definition_stage)
+        return next_backend.DEFAULT_PROG_TRANSFORMS.func_to_past(self.definition_stage)
```

Comment:
I should have asked this question, in the previous refactoring: what's happening here? Can you document it? What's the case when the `DEFAULT_PROG_TRANSFORMS` should be used?